robolectric/src/test/java/org/robolectric/shadows/ShadowPendingIntentTest.java

```diff
@@ -75,165 +94,409 @@ public void getActivities_withBundle_shouldCreateIntentForBroadcast() throws Exc
   }
 
   @Test
-  public void getService_shouldCreateIntentForBroadcast() throws Exception {
+  public void getService_shouldCreateIntentForBroadcast() {
     Intent intent = new Intent();
-    PendingIntent pendingIntent = PendingIntent.getService(RuntimeEnvironment.application, 99, intent, 100);
+    PendingIntent pendingIntent = PendingIntent.getService(context, 99, intent, 100);
+
     ShadowPendingIntent shadow = shadowOf(pendingIntent);
     assertThat(shadow.isActivityIntent()).isFalse();
     assertThat(shadow.isBroadcastIntent()).isFalse();
     assertThat(shadow.isServiceIntent()).isTrue();
     assertThat(intent).isEqualTo(shadow.getSavedIntent());
-    assertThat((Context) RuntimeEnvironment.application).isEqualTo(shadow.getSavedContext());
+    assertThat(context).isEqualTo(shadow.getSavedContext());
     assertThat(shadow.getRequestCode()).isEqualTo(99);
     assertThat(shadow.getFlags()).isEqualTo(100);
   }
 
   @Test
+  public void getActivities_nullIntent() {
+    try {
+      PendingIntent.getActivities(context, 99, null, 100);
+      fail("Expected NullPointerException when creating PendingIntent with null Intent[]");
+    } catch (NullPointerException ignore) {
+      // expected
+    }
+  }
+
+  @Test
+  public void getActivities_withBundle_nullIntent() {
+    try {
+      PendingIntent.getActivities(context, 99, null, 100, Bundle.EMPTY);
+      fail("Expected NullPointerException when creating PendingIntent with null Intent[]");
+    } catch (NullPointerException ignore) {
+      // expected
+    }
+  }
+
+  @Test
   public void send_shouldFillInIntentData() throws Exception {
     Intent intent = new Intent();
-    Activity context = new Activity();
-    PendingIntent forActivity = PendingIntent.getActivity(context, 99, intent, 100);
+    Context context = new Activity();
+    PendingIntent pendingIntent = PendingIntent.getActivity(context, 99, intent, 100);
+
+    Activity otherContext = new Activity();
+    Intent fillIntent = new Intent().putExtra("TEST", 23);
+    pendingIntent.send(otherContext, 0, fillIntent);
+
+    Intent i = shadowOf(otherContext).getNextStartedActivity();
+    assertThat(i).isNotNull();
+    assertThat(i).isSameAs(intent);
+    assertThat(i.getIntExtra("TEST", -1)).isEqualTo(23);
+  }
+
+  @Test
+  public void send_shouldFillInLastIntentData() throws Exception {
+    Intent[] intents = {new Intent("first"), new Intent("second")};
+    Context context = new Activity();
```

Comment:
Robolectric.setupActivity() - may want to make it a field initialized in setup() ?